protocols/request-response/tests/ping.rs

```diff
@@ -38,6 +38,39 @@ use rand::{self, Rng};
 use std::{io, iter};
 use std::{collections::HashSet, num::NonZeroU16};
 
+/// Check if is_response_outbound works properly
+#[test]
+fn offline_peers() {
+    let ping = Ping("ping".to_string().into_bytes());
+    let offline_peer = PeerId::random();
+
+    let protocols = iter::once((PingProtocol(), ProtocolSupport::Full));
+    let cfg = RequestResponseConfig::default();
+
+    let (peer1_id, trans) = mk_transport();
+    let ping_proto1 = RequestResponse::new(PingCodec(), protocols.clone(), cfg.clone());
+    let mut swarm1 = Swarm::new(trans, ping_proto1, peer1_id.clone());
+
+    let request_id1 = swarm1.send_request(&offline_peer, ping.clone());
+
+    let peer1 = async move {
+        match swarm1.next().await {
+            RequestResponseEvent::OutboundFailure{peer, request_id: req_id, error: _error} => {
+                assert_eq!(&offline_peer, &peer);
+                assert_eq!(req_id, request_id1);
+            },
+            e => panic!("Peer: Unexpected event: {:?}", e),
+        }
+
+        let request_id2 = swarm1.send_request(&offline_peer, ping.clone());
+                
+        assert!(!swarm1.is_pending_outbound(&offline_peer, &request_id1));
+        assert!(swarm1.is_pending_outbound(&offline_peer, &request_id2));
+    };
+    
+    let () = async_std::task::block_on(peer1);
```

Comment:
```suggestion
    match futures::executor::block_on(swarm1.next()) {
        RequestResponseEvent::OutboundFailure{peer, request_id: req_id, error: _error} => {
            assert_eq!(&offline_peer, &peer);
            assert_eq!(req_id, request_id1);
        },
        e => panic!("Peer: Unexpected event: {:?}", e),
    }

    let request_id2 = swarm1.send_request(&offline_peer, ping.clone());

    assert!(!swarm1.is_pending_outbound(&offline_peer, &request_id1));
    assert!(swarm1.is_pending_outbound(&offline_peer, &request_id2));
```

This should make it a bit more readable. What do you think?